### PR TITLE
Lvnodcdxi 67

### PR DESCRIPTION
**Pull request by Kindest13**
tested filterToggler and half of blocksReducer

### CODE REVIEW

File: src/client/components/filterToggler/filterToggler.test.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        name: 'Ring Modulator',
+        isVisible: false,
+      },
+    ],
+  };
+
+  it('should render filterToggler element', () => {
+    const nextProps = {
+      ...props,
+    };
```
Comment (NadiiaChorna) [inline src/client/components/filterToggler/filterToggler.test.js]: Why do you need here this "nextProps" constant? You can use just your "props" constant, can't you?

File: src/client/components/filterToggler/filterToggler.test.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    const wrapper = shallow(<FilterToggler {...nextProps} />);
+    expect(wrapper.find('.modesElement')).toHaveLength(1);
+  });
+
+  it('should render filterToggler properly', () => {
+    const nextProps = {
+      ...props,
+    };
+    const wrapper = shallow(<FilterToggler {...nextProps} />);
```
Comment (NadiiaChorna) [inline src/client/components/filterToggler/filterToggler.test.js]: Why do you need here this "nextProps" constant? You can use just your "props" constant, can't you?

Comment (Kindest13) [reply src/client/components/filterToggler/filterToggler.test.js]: done

**Review by mr-redFox — DISMISSED** [review]

File: src/client/components/filterToggler/filterToggler.js [hunk 3]
```diff
@@ -1,29 +1,20 @@
...
-      removeSourceFilters(audioData.sound);
-      attachFiltersToSource(audioData.voice);
+    setToggler(!toggler);
+    if (toggler) {
+      removeSourceFilters(audioData.voice, blocksData);
+      attachFiltersToSource(audioData.sound, blocksData);
     } else {
-      removeSourceFilters(audioData.voice);
-      attachFiltersToSource(audioData.sound);
+      removeSourceFilters(audioData.sound, blocksData);
```
Comment (GeFeSTTS) [inline src/client/components/filterToggler/filterToggler.js]: U can destruct sound and voice from audioData, like const { audioData: { sound, voice }, blocksData } = props;

File: src/client/components/filterToggler/filterToggler.test.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  });
+
+  it('should invoke onToggler function that attach voice filters', () => {
+    const wrapper = shallow(<FilterToggler {...props} />);
+    wrapper.find('#toggler').simulate('click');
+    expect(props.audioData.voice.addEffect).toHaveBeenCalled();
+  });
+
+  it('should invoke onToggler function that attach sound filters', () => {
+    const wrapper = shallow(<FilterToggler {...props} />);
```
Comment (tonypntsrn) [inline src/client/components/filterToggler/filterToggler.test.js]: You could add this to beforeEach and remove from every test

File: src/client/components/filterToggler/filterToggler.js (hunk 3 above)
Comment (Kindest13) [reply src/client/components/filterToggler/filterToggler.js]: done

File: src/client/components/filterToggler/filterToggler.test.js (hunk 4 above)
Comment (Kindest13) [reply src/client/components/filterToggler/filterToggler.test.js]: done

**Review by tonypntsrn — APPROVED** [review]

**Review by GeFeSTTS — APPROVED** [review]

**Pull request merged**